third_party/airflow/armada/operators/armada.py

```diff
@@ -77,57 +70,81 @@ def __init__(
         """
         Executes the Armada Operator.
-
         Runs an Armada job and calls the job_service_client for polling.
 
-        :param context: The airflow context.
-
+        :param context: The Airflow context.
         :return: None
         """
         # Health Check
         health = self.job_service.health()
         if health.status != jobservice_pb2.HealthCheckResponse.SERVING:
-            armada_logger.warn("Armada Job Service is not health")
-        # This allows us to use a unique id from airflow
-        # and have all jobs in a dag correspond to same jobset
-        job_set_id = context["run_id"]
+            armada_logger.warn("Armada Job Service is not healthy")
+
+        # This allows us to use a unique id from Airflow
+        # and have all jobs in a DAG correspond to the same jobset
+        self.job_set_id = context["run_id"]
         job = self.armada_client.submit_jobs(
             queue=self.armada_queue,
-            job_set_id=job_set_id,
+            job_set_id=self.job_set_id,
             job_request_items=annotate_job_request_items(
                 context, self.job_request_items
             ),
         )
 
         try:
-            job_id = job.job_response_items[0].job_id
+            self.job_id = job.job_response_items[0].job_id
         except Exception:
-            raise AirflowException("Armada has issues submitting job")
+            raise AirflowException("Armada encountered issues submitting the job")
 
-        armada_logger.info("Running Armada job %s with id %s", self.name, job_id)
+        armada_logger.info("Running Armada job %s with id %s", self.name, self.job_id)
 
-        lookout_url = self._get_lookout_url(job_id)
-        if len(lookout_url) > 0:
+        lookout_url = self._get_lookout_url(self.job_id)
+        if lookout_url:
             armada_logger.info("Lookout URL: %s", lookout_url)
 
         job_state, job_message = search_for_job_complete(
             job_service_client=self.job_service,
             armada_queue=self.armada_queue,
-            job_set_id=job_set_id,
+            job_set_id=self.job_set_id,
             airflow_task_name=self.name,
-            job_id=job_id,
+            job_id=self.job_id,
             poll_interval=self.poll_interval,
         )
+
         armada_logger.info(
             "Armada Job finished with %s and message: %s", job_state, job_message
         )
-        airflow_error(job_state, self.name, job_id)
+        airflow_error(job_state, self.name, self.job_id)
 
     def _get_lookout_url(self, job_id: str) -> str:
         if self.lookout_url_template is None:
             return ""
         return self.lookout_url_template.replace("<job_id>", job_id)
+
+    def on_kill(self) -> None:
+        """
+        Stops the JobService from listening to the JobSet and cancels the jobs.
+
+        :return: None
+        """
+        try:
+            if self.job_set_id && self.queue:
+                
```

Comment:
Can you remove the newlines?

Reply:
is it ok now @kannon92 ? Still ci failing .

Reply:
Two things:

1) I think there is a formatting issue and it is failing.  Try running `tox -e format-code`.  You will have to get a python environment.  See the github workflow for how to set this up.
2) You need to regenerate the docs since you changed the doc-strings.